src/Bladeburner/Bladeburner.ts

```diff
@@ -750,31 +751,26 @@ export class Bladeburner {
     }
   }
 
+  private getActionCasualties(action: Operation | BlackOperation, success: boolean) {
+    const severity = success ? CasualtyFactor.LOW_CASUALTIES : CasualtyFactor.HIGH_CASUALTIES;
+    const casualties = new TeamCasualties(severity, action.teamCount, this.sleeveSize);
+    return casualties.rollOutcome(getRandomIntInclusive, this);
+  }
+
+  public killSupportingSleeves(sleeveDeaths: number) {
+    const sup = Player.sleeves.filter((x) => isSleeveSupportWork(x.currentWork));
+    const damagedIndices = [...Array(Math.min(sup.length, sleeveDeaths)).keys()].sort(() => 0.5 - Math.random());
+    damagedIndices.forEach((idx) => sup[idx].kill());
+  }
```

Comment:
I would love to see this also excluded to TeamCasualities.ts
Maybe provide a wrapper to get the sleeves that are doing support work still from bladeburner, meaning keeping this 'bad' code out: 
`Player.sleeves.filter((x) => isSleeveSupportWork(x.currentWork));` and just provide SleeveTeamMembers or getSleeveTeamMembers to the TeamCasualities.rollOutcome function.

I hope the god singleton object Player will get refactored later to have some less responsibilities xD.

Reply:
These two are tricky. I'm not sure if the heuristic of picking X random sleeves from the player should be that deep in casualties.

I agree that there should be a heuristic to highlight that the sleeves killed **are random**, ie. killRandomSleeves, not just killXSleevesInOrder, but for the implementation I'd rather keep in on BB or Player till that mess gets moved somewhere else (I have a feeling I'll be the one doing that anyway).

Reply:
What's the problem with that? I may as well move this to SleevePerson or Person.